app/models/manageiq/providers/openstack/infra_manager.rb

```diff
@@ -73,4 +74,36 @@ def supports_authentication?(authtype)
   def self.event_monitor_class
     ManageIQ::Providers::Openstack::InfraManager::EventCatcher
   end
+
+  def clouds
+    overclouds = ManageIQ::Providers::Openstack::CloudManager.find(:all, :conditions => ["id != ? and provider_id = ?", id, provider_id])
+  end
+
+  #
+  # Statistics
+  #
+
+  def cloud_cinder_disk_usage
+    cinder_used = 0
+    clouds.each do |cloud|
+      cinder_used += cloud.cinder_disk_usage
+    end
+    cinder_used
+  end
+
+  def cloud_swift_disk_usage
+    replicas = OrchestrationStackParameter.joins(:stack).where("orchestration_stacks.ems_id = ? AND orchestration_stack_parameters.name = 'SwiftReplicas'", id).first.value.to_i
+    object_storage_count = OrchestrationStackParameter.joins(:stack).where("orchestration_stacks.ems_id = ? AND orchestration_stack_parameters.name = 'ObjectStorageCount'", id).first.value.to_i
```

Comment:
```ruby
replicas = orchestration_stacks.map { |stack| stack.parameters.find_by(name: 'SwiftReplicas') }.compact.first.value.to_i
```

But is it really okay that we just grab the value from an arbitrarily chosen one of the orchestration stacks linked to this manager?

If they're really interchangeable, we can simplify things by adding an association:

```ruby
has_many :orchestration_stack_parameters, :through => :orchestration_stacks, :source => :parameters
# ..
replicas = orchestration_stack_parameters.find_by(name: 'SwiftReplicas').value.to_i
```